File: src/Python/744.NextGreatestLetter.py

class Solution:
    def nextGreatestLetter(self, letters, target):
        """
        :type letters: List[str]
        :type target: str
        :rtype: str
        """
        for l in letters:
            if l > target:
                return l
        return letters[0]


if __name__ == '__main__':
    solution = Solution()
    print(solution.nextGreatestLetter(["c", "f", "j"], "a"))
    print(solution.nextGreatestLetter(["c", "f", "j"], "c"))
    print(solution.nextGreatestLetter(["c", "f", "j"], "k"))
else:
    pass

File: src/Java/701-800/744.NextGreatestLetter.java

import java.awt.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.LinkedList;
import java.util.Queue;
import java.util.HashMap;
import java.util.HashSet;

class Solution {
    public char nextGreatestLetter(char[] letters, char target) {
        int n = letters.length;
        if (target >= letters[n - 1]) {
            return letters[0];
        }
        int l = 0;
        int r = n - 1;
        while (l <= r) {
            int mid = l + (r - l) / 2;
            if (letters[mid] > target)
                r = mid - 1;
            else if(letters[mid] < target)
                l = mid + 1;
            else{
                while (letters[mid] == target)
                    mid += 1;
                return letters[mid];
            }
        }
        return letters[l];
    }
}
